### ISSUE TITLE
ValueError occurs when mkdocs serve

### ISSUE THREAD

**Issue by tasanuma** [squidfunk/mkdocs-material#4885]
### Bug description

When starting the server by `mkdocs serve` with mkdcoks-9.0.5, the following error occurred.
```
mkdocs serve --livereload
...
ERROR    -  Error building page 'test.md': '<li>' is not in list
Traceback (most recent call last):
  File "/.../venv/mkdocs/bin/mkdocs", line 10, in <module>
    ...
  File "/.../venv/mkdocs/lib/python3.8/site-packages/material/plugins/search/plugin.py", line 448, in handle_endtag
    index = data.index(f"<{tag}>")
ValueError: '<li>' is not in list
```

After reverting https://github.com/squidfunk/mkdocs-material/commit/68257348b883ca6bccc514e1afdad8852a458646, the error doesn't occur.

### Related links

- https://squidfunk.github.io/mkdocs-material/creating-your-site/#previewing-as-you-write
- https://github.com/squidfunk/mkdocs-material/commit/68257348b883ca6bccc514e1afdad8852a458646

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/10449404/example.zip)

### Steps to reproduce

1. an article includes a list of a heading.
```
- #foo
```

2. starting the server

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/bug-report/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) or [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/bug-report/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/bug-report/reproduction/).

**Comment by LandonTClipp** [comment]
This is also occurring for me. The issue happened with a doc that looked like this:

```
1. This is some text

    List Title
    -------

    List title text

2. This is more text
```

**Comment by squidfunk** [comment]
Fixed in e7013e66a. This was just a minor issue when exiting sections as described in https://github.com/squidfunk/mkdocs-material/issues/4824#issuecomment-1374762698 which was caused by an invalid check. We now set exited section depth to infinity, which will exclude them from further consideration when adding parsed tags and text. I've also verified that all cases in #4824 still work.

**Comment by squidfunk** [comment]
Released as part of [9.0.6](https://github.com/squidfunk/mkdocs-material/releases/tag/9.0.6).

**Comment by tasanuma** [comment]
@squidfunk Sorry my zip is not enough, and thanks for the quick fix!
